src/run.js

```diff
@@ -0,0 +1,76 @@
+const { spawn } = require('child_process');
+const { ensureRelayer } = require('./download');
+const { relayHub } = require('./data');
+const { deployRelayHub, registerRelay } = require('./helpers');
+const sleep = require('sleep-promise');
+const tmp = require('tmp');
+const { chunk } = require('lodash');
+
+async function runRelayer({ detach, workdir, devMode, ethereumNodeURL, gasPricePercent, port, relayUrl, relayHubAddress, quiet }) {
+  // Download relayer if needed
+  const binPath = await ensureRelayer();
+
+  // Create tmp dir
+  const workingDir = workdir || tmp.dirSync({ unsafeCleanup: true }).name;
+
+  // Build args
+  const args = [];
+  if (ethereumNodeURL) args.push('-EthereumNodeUrl', ethereumNodeURL);
+  if (relayHubAddress) args.push('-RelayHubAddress', relayHubAddress);
+  args.push('-Port', getPort({ relayUrl, port }));
+  args.push('-Url', getUrl({ relayUrl, port }));
+  args.push('-GasPricePercent', gasPricePercent || 0);
+  args.push('-Workdir', workingDir);
+  if (devMode !== false) args.push('-DevMode');
+  
+  // Run it!
+  console.error(`Starting relayer\n${binPath}\n${chunk(args, 2).map(arr => ' ' + arr.join(' ')).join('\n')}`);
+  return spawn(binPath, args, {
+    stdio: (quiet || detach) ? 'ignore' : 'inherit',
+    detached: !!detach
+  });
+}
+
+async function runAndRegister(web3, opts={}) {
+  const { from } = opts;
+  let { relayHubAddress } = opts;
+
+  // Deploy relay hub if needed
+  if (!relayHubAddress || relayHubAddress === relayHub.address) {
+    relayHubAddress = await deployRelayHub(web3, { from });
+  }
+
+  // Start running relayer and register it
+  const process = await runRelayer({ ...opts, relayHubAddress });
```

Comment:
`process` is not a good name because it conflicts with Node's `process` global variable. It threw me off to see `process.kill()` below.